script/scaffold/templates/config_flow/integration/config_flow.py

```diff
@@ -1,4 +1,6 @@
 """Config flow for NEW_NAME integration."""
+from __future__ import annotations
```

Comment:
I don't see a reason in this file that requires the future import?

Reply:
With this present, `pyupgrade` will automatically handle all future typing updates. Once it is the default in `py310` it can just as easily remove it.
Adding the import in the templates will make future runs of `python-typing-update` more or less unnecessary.

Since the `__future__` import is basically a flag there shouldn't be  much of a performance penalty .

Reply:
The automatic update will be added with #48089

Reply:
It code that is not used in the file. It should not be there.

Reply:
Alternatively, we could start adding typing as a good starting point. But I'm not sure if that is adding a level of complexity that is wished for.

Reply:
My idea behind that was that if someone uses the template and later adds typing the reviewer won't have to check for `Union`, `Optional` and so on. It can just be handled by `pyupgrade`.

That however does only happen if the import is present in the first place. Someone who doesn't know much about typing probably won't add it on their own.

--

The way my tool works is basically just that:
1. It adds `from __future__  import annotation` to every file
2. Runs the new version of pyupgrade
3. Checks if any import was removed
4. If not, reverts the change.

So in theory adding the import to every file would remove the need for it.

Reply:
I get how it works :)

Still think we should not do unused imports.

Reply:
I've removed them now. Just pushed a new commit.

Reply:
I'm seriously considering adding typing to all the templates (for another PR so people can kill it 😉 )